core/controllers/base_test.py

```diff
@@ -707,38 +707,39 @@ def test_downloadable(self):
         self.assertEqual(response.content_type, 'text/plain')
 
 
+class SessionBeginHandlerTests(test_utils.GenericTestBase):
+    """Tests for session handler."""
+
+    def test_get(self):
+        self.get_html_response('/session_begin', expected_status_int=200)
+
+
 class LogoutPageTests(test_utils.GenericTestBase):
+    """Tests for logout handler."""
 
     def test_logout_page(self):
-        """Tests for logout handler."""
         exp_services.load_demo('0')
+        self.get_html_response('/explore/0')
+
         # Logout with valid query arg. This test only validates that the login
         # cookies have expired after hitting the logout url.
-        current_page = '/explore/0'
-        self.get_html_response(current_page)
         response = self.get_html_response('/logout', expected_status_int=302)
-        expiry_date = response.headers['Set-Cookie'].rsplit('=', 1)
-        self.assertTrue(
-            datetime.datetime.utcnow() > datetime.datetime.strptime(
-                expiry_date[1], '%a, %d %b %Y %H:%M:%S GMT'))
+
+        self.assertNotIn('Set-Cookie', response.headers)
 
     def test_logout_page_with_redirect_url(self):
         exp_services.load_demo('0')
-        current_page = '/explore/0'
-        self.get_html_response(current_page)
+        self.get_html_response('/explore/0')
+
         response = self.get_html_response(
             '/logout?redirect_url=community-library', expected_status_int=302)
-        expiry_date = response.headers['Set-Cookie'].rsplit('=', 1)
 
-        self.assertTrue(
-            datetime.datetime.utcnow() > datetime.datetime.strptime(
-                expiry_date[1], '%a, %d %b %Y %H:%M:%S GMT'))
+        self.assertNotIn('Set-Cookie', response.headers)
```

Comment:
Are we ever verifying that the 'Set-Cookie' is in the headers? As-is, I can't be sure that this will ever fail when expected.

Reply:
This is tricky now that I double-check. Auth services are stubbed out in backend tests (source), and request/response headers are not modified by the stub.

This means that the logout function, which calls the stubbed `auth_services.destroy_auth_session`, has nothing valuable to "check". The best I can do is verify that the function gets called. Would that be OK?

Reply:
Updated to check that the function is called.